build.gradle

```diff
@@ -121,32 +111,75 @@ subprojects {
     publishing {
         publications {
             mavenJava(MavenPublication) {
+                artifactId = archivesBaseName
                 from components.java
-                artifact sourceJar
-                artifact javadocJar
-
-                afterEvaluate {
-                    artifactId = archivesBaseName
+                pom {
+                    name = project.name
+                    description = project.description
+                    url = 'http://ultrabrew.io'
+                    scm {
+                        connection = 'scm:git:git://github.com/ultrabrew/metrics.git'
+                        developerConnection = 'scm:git:ssh://github.com:ultrabrew/metrics.git'
+                        url = 'https://github.com/ultrabrew/metrics.git'
+                    }
+                    licenses {
+                        license {
+                            name = 'The Apache License, Version 2.0'
+                            url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                        }
+                    }
+                    developers {
+                        developer {
+                            id = 'mmannerm'
+                            name = 'Mika Mannermaa'
+                        }
+                        developer {
+                            id = 'smrutilal2'
+                            name = 'Smruti Ranjan Sahoo'
+                        }
+                        developer {
+                            id = 'lonemeow'
+                            name = 'Ilpo Ruotsalainen'
+                        }
+                        developer {
+                            id = 'manolama'
+                            name = 'Chris Larsen'
+                        }
+                        developer {
+                            id = 'arungupta'
+                            name = 'Arun Gupta'
+                        }
+                    }
                 }
             }
         }
 
         repositories {
             maven {
-                def artifactory = 'https://oss.jfrog.org/artifactory/oss-snapshot-local'
-                def bintray = 'https://api.bintray.com/maven/ultrabrew/m2/' + rootProject.name + '/;publish=1'
+                def snapshotRepository = 'https://s01.oss.sonatype.org/content/repositories/snapshots/'
+                def stagingRepository = 'https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/'
 
                 afterEvaluate {
-                    url = version.toString().endsWith('-SNAPSHOT') ? artifactory : bintray
+                    url = version.toString().endsWith('-SNAPSHOT') ? snapshotRepository : stagingRepository
```

Comment:
How does the artifact end up in public repo from the staging repo?

Reply:
It's manually done from the staging repo. It also can be automated so that every build is published to the central repo. But, I kept it manual to start with. Once we verify that it's building fine, we can automate the promotion.

Reply:
Ref: https://central.sonatype.org/publish/release/

Reply:
Ideally we'd automatically promote releases that are tagged as releases in git, I suppose.

Reply:
Yes, I will add that in my next pr may be.